Fix safe area observing for example query #124

iphone201988 opened the pull request.

twodayslate commented on ec3730/Views/Host/HostBarView.swift:
```diff
@@ -40,12 +40,7 @@ struct HostBarView: View {
-                .ignoresSafeArea(.all, edges: .horizontal)
-        )
-        .ignoresSafeArea()
+        }.ignoresSafeArea()
```
Let's just have the relevant changes in the PR

twodayslate commented:
Please rebase

iphone201988 commented:
[https://we.tl/t-OjRjdFyePK](url) already up to date

twodayslate commented on ec3730/Views/Host/HostViewSectionFocusView.swift:
```diff
@@ -10,7 +10,7 @@ struct HostViewSectionFocusView: View {
                     HostViewSectionContent(sectionModel: model, canQuery: true)
                 }
-                HostBarView(url: url, date: date)
+                HostBarView(url: url, date: date).background(Color(UIColor.systemGroupedBackground))
```
Why is this change necessary?

iphone201988 replied:
HostBarView(url: url, date: date).background(Color(UIColor.systemGroupedBackground))

[https://we.tl/t-FPltFCM832](url) Without background color , it will show like the given screenshot, there will be no background color.

twodayslate replied:
Shouldn't this just ignore the safe area?

iphone201988 replied:
[https://we.tl/t-2KPswENwxO](url) attached screenshots , in case we use only the ignore safe area , or even without that there is no background color, but if we want background color , then we need to give background color

twodayslate commented:
Does https://twitter.com/natpanferova/status/1590676836488732678?s=46&t=DunNho2WvomMLprYw9b4pw help?

techwinlabs commented:
> Does https://twitter.com/natpanferova/status/1590676836488732678?s=46&t=DunNho2WvomMLprYw9b4pw help?

If you check this [link]( https://twitter.com/natpanferova/status/1590676836488732678?s=46&t=DunNho2WvomMLprYw9b4pw) , 

Background color also given same as I implemented in snippet.

Please check these screenshots: [https://we.tl/t-ujdUL7nHMj](url) 
I have added the code as suggested, but we need to provide the background color.

If background color is not provided it will show as:  [https://we.tl/t-SS3ovvpGF7](url)

iphone201988 commented:
> Does https://twitter.com/natpanferova/status/1590676836488732678?s=46&t=DunNho2WvomMLprYw9b4pw help?

If you check this [link](https://twitter.com/natpanferova/status/1590676836488732678?s=46&t=DunNho2WvomMLprYw9b4pw) ,

Background color also given same as I implemented in snippet.

Please check these screenshots: [https://we.tl/t-ujdUL7nHMj](https://github.com/twodayslate/NetUtils/pull/url)
I have added the code as suggested, but we need to provide the background color.

If background color is not provided it will show as: [https://we.tl/t-SS3ovvpGF7](https://github.com/twodayslate/NetUtils/pull/url)

twodayslate commented:
This seemed to work for me

```
diff --git a/ec3730/Views/Host/HostBarView.swift b/ec3730/Views/Host/HostBarView.swift
index 294cbc7..33358a6 100644
--- a/ec3730/Views/Host/HostBarView.swift
+++ b/ec3730/Views/Host/HostBarView.swift
@@ -41,11 +41,6 @@ struct HostBarView: View {
                 Spacer()
             }
         }
-        .background(
-            VisualEffectView(effect: UIBlurEffect(style: .systemMaterial))
-                .ignoresSafeArea(.all, edges: .horizontal)
-        )
-        .ignoresSafeArea()
         .confirmationDialog("Information", isPresented: $showInfo) {
             Button(action: {
                 UIPasteboard.general.string = url.absoluteString
@@ -58,6 +53,10 @@ struct HostBarView: View {
                 Label("Copy Date", systemImage: "doc.on.doc")
             })
         }
+        .background(
+            VisualEffectView(effect: UIBlurEffect(style: .systemMaterial))
+                .ignoresSafeArea()
+        )
     }
 }
 
diff --git a/ec3730/Views/Host/HostViewSectionFocusView.swift b/ec3730/Views/Host/HostViewSectionFocusView.swift
index 5f2f32d..dd5cdc7 100644
--- a/ec3730/Views/Host/HostViewSectionFocusView.swift
+++ b/ec3730/Views/Host/HostViewSectionFocusView.swift
@@ -17,3 +17,9 @@ struct HostViewSectionFocusView: View {
         }
     }
 }
+
+struct HostViewSectionFocusViewPreview: PreviewProvider {
+    static var previews: some View {
+        HostViewSectionFocusView(model: WhoisXmlDnsSectionModel(), url: URL(string: "https://google.com")!, date: .now)
+    }
+}
```

twodayslate commented:
It's also used in HostResult.swift

iphone201988 commented:
Please Suggest , on which ticket we can work upon next

1.https://github.com/twodayslate/NetUtils/issues/108
Add status progress bar for source view · Issue #108 · twodayslate/[NetUtilsgithub.com](http://netutilsgithub.com/)
2.https://github.com/twodayslate/NetUtils/issues/101
Create Image enum/extension · Issue #101 · twodayslate/[NetUtilsgithub.com](http://netutilsgithub.com/)
3.https://github.com/twodayslate/NetUtils/issues/102
Create UserDefaults extensions · Issue #102 · twodayslate/[NetUtilsgithub.com](http://netutilsgithub.com/)
4.https://github.com/twodayslate/NetUtils/issues/98
Disable navigation to PingSetList if there are no ping sets · Issue #98 · twodayslate/[NetUtilsgithub.com](http://netutilsgithub.com/)
5.https://github.com/twodayslate/NetUtils/issues/82
Hide host lookup bar on scroll · Issue #82 · twodayslate/[NetUtilsgithub.com](http://netutilsgithub.com/)
6.https://github.com/twodayslate/NetUtils/issues/111

iphone201988 commented:
> It's also used in HostResult.swift

[https://we.tl/t-QxHhGQZNh6](url) This view has tabbar below , please check screenshot, If any updation required let me know.

twodayslate commented:
Can that use the safe area unset as well?
